my-app/src/app/core/services/bread-crumb/bread-crumb.service.ts

```diff
@@ -17,13 +17,11 @@ export class BreadCrumbsService {
 
   }
 
-  public getCourseEditorCrumbs(id: number): Breadcrumb[] {
+  public getCourseEditorCrumbs(course: Course): Breadcrumb[] {
```

Comment:
there should not be separate method for each page. You should parse router url and return array of path